<launch>
  <!-- main collision environment parameter -->
  <arg name="mesh_source"         default="$(env HOME)/data/primesense/mesh" 
    doc="Location of the folder that contains the mesh object files" />
  <arg name="file_extension"      default="obj" 
    doc="The extension that used by all mesh files. Supports assimp importable extension, such as: stl, obj, dae, blend" />

  <arg name="charToFind"          default="::" 
    doc="Signature characters that separates between the name of object inside of the TF name"/>
  <arg name="objectNameIdx"       default="2" 
    doc="The position of the object name after charToFind separation"/>
  <arg name="tableTF"         default="ar_marker_2" 
    doc="The TF name for the table" />
  <arg name="defineParent"        default="true" 
    doc="Boolean variable to set whether we use predefined parent name for collision object TF header or not" />
  <arg name="parentFrameName"     default="base_link"
    doc="Predefined collision object TF header" />
  <arg name="renewTable"          default="true" />

  <arg name="useTableWall"           default="true" doc="Add a wall on back,left, and right edge of the table."/>
  <arg name="useBaseLinkWall"        default="true" doc="Add a wall on back,left, and right edge of the baseLink."/>

  <arg name="baseLinkName"           default="base_link" doc="The name of base link used for generating robot ground"/>
  <arg name="useBaseLinkGround"      default="true" doc="Add robot ground"/>
  <arg name="tableSize"              default="1.0" doc="Table and wall size in meters" />
  <arg name="baseLinkWallDistance"   default="1.0" doc="Baselink to  wall distance in meters" />

  <!-- Enable for debugging purpose -->
  <arg name="debug"               default="false" />

  <node pkg="moveit_collision_environment" type="moveit_collision_environment" name="planningSceneGenerator">  
  <!-- moveit_collision_environment arg pass -->
    <param name="mesh_source"     type="str"  value="$(arg mesh_source   )" />
    <param name="file_extension"  type="str"  value="$(arg file_extension)" />

    <param name="charToFind"      type="str"  value="$(arg charToFind    )" />
    <param name="objectNameIdx"   type="int"  value="$(arg objectNameIdx )" />

    <param name="tableTFname"     type="str"  value="$(arg tableTF   )" />
    <param name="renewTable"      type="bool" value="$(arg renewTable    )" />


    <param name="useTableWall"    type="bool" value="$(arg useTableWall )" />
    <param name="useBaseLinkGround"    type="bool" value="$(arg useBaseLinkGround  )" />
    <param name="baseLinkName"    type="str" value="$(arg baseLinkName )" />

    <param name="tableSize"   type="double"  value="$(arg tableSize )" />
    <param name="baseLinkWallDistance"   type="double"  value="$(arg baseLinkWallDistance )" />

    <param name="defineParent"    type="bool" value="$(arg defineParent  )" />
    <param name="parentFrameName" type="str"  value="$(arg parentFrameName)" />


    <param name="debug"           type="bool" value="$(arg debug)" />
  </node>

</launch>
